SPECS/xdp-tools/xdp-tools.spec

```diff
@@ -0,0 +1,241 @@
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Name:             xdp-tools
+Version:          1.4.1
+Release:          2%{?dist}
+Summary:          Utilities and example programs for use with XDP
+%global _soversion 1.4.0
+
+License:          GPL-2.0-only
+URL:              https://github.com/xdp-project/%{name}
+Source0:          https://github.com/xdp-project/xdp-tools/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+
+%global azl 3
+BuildRequires:    libbpf-devel
+BuildRequires:    elfutils-libelf-devel
+BuildRequires:    zlib-devel
+BuildRequires:    libpcap-devel
+BuildRequires:    clang >= 10.0.0
+BuildRequires:    llvm >= 10.0.0
+BuildRequires:    make
+BuildRequires:    gcc
+BuildRequires:    pkgconfig
+BuildRequires:    m4
+# Azlinux does not have emacs. Docs will not be built.
+%if !%{azl}
+BuildRequires:    emacs-nox
+%endif
+# Azlinux does not have wireshark, required for tests. Tests will not be run.
+%if !%{azl}
+BuildRequires:    wireshark-cli
+%endif
+
+%ifnarch i686
+BuildRequires:    bpftool
+%endif
+
+# Always keep xdp-tools and libxdp packages in sync
+Requires:         libxdp = %{version}-%{release}
+
+# find-debuginfo produces empty debugsourcefiles.list
+# disable the debug package to avoid rpmbuild error'ing out because of this
+%global debug_package %{nil}
+%global _hardened_build 1
+# To avoid install errors of the kind:
+# /bin/strip: Unable to recognise the format of the input file
+# `/usr/src/mariner/BUILDROOT/xdp-tools-1.4.1-1.azl3.x86_64/usr/lib/bpf/xdpdump_xdp.o'
+%if %{azl}
+%global __strip /bin/true
+%endif
```

Comment:
Maybe, we move these to the top of the spec as traditionally we have put macros at the top of the spec :)

Reply:
This is nearly a copy of the upstream spec file available as part of the xdp-tools sources https://github.com/xdp-project/xdp-tools/blob/master/packaging/rpm/xdp-tools.spec so I left the macros at the same position as found there.
Since it doesn't impact the readability much, I will leave it as is.
TFTR.

Reply:
I agree with the rationale :)